common/scripts/access-control/flexible-server/Grant-FlexibleServerAccess.ps1

```diff
@@ -0,0 +1,153 @@
+    [void]$builder.Append('     END IF; ')
+    [void]$builder.Append("     IF NOT EXISTS (SELECT 1 FROM pgaadauth_list_principals(false) WHERE rolname='$ServiceMIName') THEN ")
+    [void]$builder.Append("         RAISE NOTICE 'CREATING PRINCIPAL FOR MANAGED IDENTITY:$ServiceMIName';")
+    [void]$builder.Append("         PERFORM pgaadauth_create_principal('$ServiceMIName', false, false); ");
+    [void]$builder.Append("         RAISE NOTICE 'PRINCIPAL FOR MANAGED IDENTITY CREATED:$ServiceMIName';")
+    [void]$builder.Append('     ELSE ')
+    [void]$builder.Append("         RAISE NOTICE 'PRINCIPAL FOR MANAGED IDENTITY ALREADY EXISTS:$ServiceMIName';")
+    [void]$builder.Append('     END IF; ')
+    [void]$builder.Append("     EXECUTE ( 'GRANT CONNECT ON DATABASE `"$PostgresDatabase`" TO `"$TeamMIName`"' );")
+    [void]$builder.Append("     EXECUTE ( 'GRANT CONNECT ON DATABASE `"$PostgresDatabase`" TO `"$ServiceMIName`"' );")
+    [void]$builder.Append("     RAISE NOTICE 'GRANTED CONNECT TO DATABASE';")
+    [void]$builder.Append(" EXCEPTION ")
+    [void]$builder.Append("     WHEN OTHERS THEN  ")
+    [void]$builder.Append("         RAISE EXCEPTION 'ERROR DURING PRINCIPAL CREATION/GRANT CONNECT: %', SQLERRM; ")
+    [void]$builder.Append(' END $$' )
+    return $builder.ToString()
+}
+
+function Get-SQLScriptToGrantAllPermissions {
+    [System.Text.StringBuilder]$builder = [System.Text.StringBuilder]::new()
+    [void]$builder.Append("GRANT ALL ON SCHEMA public TO `"$TeamMIName`";")
+    [void]$builder.Append("GRANT ALL ON ALL TABLES IN SCHEMA public TO `"$TeamMIName`";")
+    [void]$builder.Append("GRANT ALL ON ALL SEQUENCES IN SCHEMA public TO `"$TeamMIName`";")
+    [void]$builder.Append("GRANT ALL ON ALL FUNCTIONS IN SCHEMA public TO `"$TeamMIName`";")
+    [void]$builder.Append("GRANT ALL ON ALL PROCEDURES IN SCHEMA public TO `"$TeamMIName`";")
+    return $builder.ToString()
+}
+
+try {
+    [System.IO.DirectoryInfo]$moduleDir = Join-Path -Path $WorkingDirectory -ChildPath "common/scripts/modules/psql"
+    Write-Debug "${functionName}:moduleDir.FullName=$($moduleDir.FullName)"
+    Import-Module $moduleDir.FullName -Force
+
+    Write-Host "Connecting to Azure..."
+    $null = Connect-AzAccount -ServicePrincipal -ApplicationId $PlatformMIClientId -FederatedToken $(Get-Content $PlatformMIFederatedTokenFile -raw) -Tenant $PlatformMITenantId -Subscription $PlatformMISubscriptionId
+    $null = Set-AzContext -Subscription $SubscriptionName
+    Write-Host "Connected to Azure and set context to '$SubscriptionName'"
+
+    Write-Host "Acquiring Access Token..."
+    $accessToken = Get-AzAccessToken -ResourceUrl "https://ossrdbms-aad.database.windows.net"
+    $ENV:PGPASSWORD = $accessToken.Token
+    Write-Host "Access Token Acquired"
+
+    [string]$command = Get-SQLScriptToCreatePrincipal
+    Write-Debug "${functionName}:command=$command"
+    
+    [System.IO.FileInfo]$tempFile = [System.IO.Path]::GetTempFileName()
```

Comment:
Define meaningful names for tempFile and tempFile1 ?

Reply:
@ppotturi done